test/harness.ts

```diff
@@ -81,7 +81,7 @@ export function hideFromStackTrace(block: CallableFunction) {
 }
 
 export function tempDirWithFiles(basename: string, files: Record<string, string>) {
-  const dir = fs.mkdtempSync(path.join(os.tmpdir(), basename + "_"));
+  const dir = fs.mkdtempSync(path.join(fs.realpathSync(os.tmpdir()), basename + "_"));
```

Comment:
Looks like there are some new `test-test.test.ts` failures, might be from this change?

Reply:
1. tests on `bun-darwin-aarch64` weren't even running before due to `sudo`, which was fixed recently: #3254
2. `test-test.test.ts` is also failing in a separate PR: https://github.com/oven-sh/bun/actions/runs/5248738253/jobs/9480987433?pr=3288#step:6:7